syntax = "proto3";

package orchestration;

import "placeholder.proto";

// Specification for Python-class based executables.
message PythonClassExecutableSpec {
  // The full class path of the executable.
  string class_path = 1;
  // Extra flags for running the Python executable.
  repeated string extra_flags = 2;
}

// Specification for Beam based executables.
message BeamExecutableSpec {
  // Nested python executor spec.
  PythonClassExecutableSpec python_executor_spec = 1;
  // Args for Apache Beam pipeline.
  repeated string beam_pipeline_args = 2;
}

// Specification for Container based executables.
// This executable resembles the spec defined by TemplatedExecutorContainerSpec.
message ContainerExecutableSpec {
  // The uri of the container image.
  string image = 1;
  // The commands of the container.
  repeated PlaceholderExpression commands = 2;
  // The args of the container entrypoint.
  repeated PlaceholderExpression args = 3;
}